superset/sql_parse.py

```diff
@@ -206,14 +275,91 @@ def __init__(self, sql_statement: str, strip_comments: bool = False):
     @property
     def tables(self) -> set[Table]:
         if not self._tables:
-            for statement in self._parsed:
-                self._extract_from_token(statement)
-
-            self._tables = {
-                table for table in self._tables if str(table) not in self._alias_names
-            }
+            self._tables = self._extract_tables_from_sql()
         return self._tables
 
+    def _extract_tables_from_sql(self) -> set[Table]:
+        """
+        Extract all table references in a query.
+
+        Note: this uses sqlglot, since it's better at catching more edge cases.
+        """
+        try:
+            statements = parse(self.sql, dialect=self._dialect)
+        except Exception:  # pylint: disable=broad-exception-caught
```

Comment:
We can catch `ParseError` from `sqlglot.errors`. Also, I am thinking we should capture an exception here rather than a warning?

Reply:
Good point, a more specific exception is better.

I'm also not a fan of the warning, but the current behavior today is to return an empty set if the parsing fails. I wanted to keep the existing behavior because changing the parsing is already a big change.